gradoop-temporal/src/main/java/org/gradoop/temporal/model/impl/TemporalGraphCollection.java

```diff
@@ -227,14 +234,50 @@ public GraphCollection toGraphCollection() {
    * {@link GraphCollection} with default values for the temporal attributes.
    *
    * @param graphCollection the existing graph collection instance
-   * @return a temporal graph colection with default temporal values
+   * @return a temporal graph collection with default temporal values
    * @see TemporalGraphCollectionFactory#fromNonTemporalGraphCollection(BaseGraphCollection)
    */
   public static TemporalGraphCollection fromGraphCollection(GraphCollection graphCollection) {
     return TemporalGradoopConfig.fromGradoopFlinkConfig(graphCollection.getConfig())
       .getTemporalGraphCollectionFactory().fromNonTemporalGraphCollection(graphCollection);
   }
 
+  /**
+   * Convenience API function to create a {@link TemporalGraphCollection} from an existing
+   * {@link GraphCollection} with valid times depending on the 3 given {@link TimeIntervalExtractor} functions
+   *
+   * @param graphCollection     the existing graph collection instance
+   * @param graphTimeExtractor  the time extractor function for the Graph Heads in the
+   *                            graph collection instance
+   * @param vertexTimeExtractor the time extractor function for the vertices in the graph collection instance
+   * @param edgeTimeExtractor   the time extractor function for the edges in the graph collection instance
+   * @return a temporal graph collection with new valid time values
+   * @see TemporalGraphCollectionFactory#fromNonTemporalGraphCollection(BaseGraphCollection)
+   */
+  public static TemporalGraphCollection fromGraphCollection(GraphCollection graphCollection,
+    TimeIntervalExtractor<EPGMGraphHead> graphTimeExtractor,
+    TimeIntervalExtractor<EPGMVertex> vertexTimeExtractor,
+    TimeIntervalExtractor<EPGMEdge> edgeTimeExtractor) {
+
+    TemporalGradoopConfig temporalGradoopConfig =
+      TemporalGradoopConfig.fromGradoopFlinkConfig(graphCollection.getConfig());
+
+    DataSet<TemporalGraphHead> temporalGraphHeadsDataSet = graphCollection.getGraphHeads().map(
+      new GraphHeadToTemporalGraphHead<>(
+        temporalGradoopConfig.getTemporalGraphFactory().getGraphHeadFactory(), graphTimeExtractor));
+
+    DataSet<TemporalVertex> temporalVertexDataSet = graphCollection.getVertices().map(
+      new VertexToTemporalVertex<>(temporalGradoopConfig.getTemporalGraphFactory().getVertexFactory(),
+        vertexTimeExtractor));
+
+    DataSet<TemporalEdge> temporalEdgeDataSet = graphCollection.getEdges().map(
+      new EdgeToTemporalEdge<>(temporalGradoopConfig.getTemporalGraphFactory().getEdgeFactory(),
+        edgeTimeExtractor));
+
+    return temporalGradoopConfig.getTemporalGraphCollectionFactory()
+      .fromDataSets(temporalGraphHeadsDataSet, temporalVertexDataSet, temporalEdgeDataSet);
```

Comment:
Same here, the `TemporalGraphCollectionFactory` has a method called `fromNonTemporalDataSets` too, which can be used in a similar way.